Die rather than just warn if the AFS cache fails any of its sanity checks,
diagnose use of tmpfs, and reject any file system type other than ext2 or
ext3.  Submitted upstream as #20947.

--- openafs/src/afsd/afsd.c	(revision 1849)
+++ openafs/src/afsd/afsd.c	(local)
@@ -1069,7 +1069,11 @@
 	    return "cannot use reiserfs as cache partition";
 	} else if  (statfsbuf.f_type == 0x58465342) { /* XFS_SUPER_MAGIC */
 	    return "cannot use xfs as cache partition";
-	}
+	} else if (statfsbuf.f_type == 0x01021994) {    /* TMPFS_SUPER_MAGIC */
+            return "cannot use tmpfs as cache partition";
+        } else if (statfsbuf.f_type != 0xEF53) {
+            return "must use ext2 or ext3 for cache partition";
+        }
     }
 #endif
 
@@ -1665,8 +1669,10 @@
     sprintf(fullpn_VFile, "%s/", cacheBaseDir);
     vFilePtr = fullpn_VFile + strlen(fullpn_VFile);
 
-    if  (!(cacheFlags & AFSCALL_INIT_MEMCACHE) && (fsTypeMsg = CheckCacheBaseDir(cacheBaseDir))) {
-	printf("%s: WARNING: Cache dir check failed (%s)\n", rn, fsTypeMsg);
+    if (!(cacheFlags & AFSCALL_INIT_MEMCACHE)
+        && (fsTypeMsg = CheckCacheBaseDir(cacheBaseDir))) {
+	printf("%s: ERROR: Cache dir check failed (%s)\n", rn, fsTypeMsg);
+        exit(1);
     }
 #if 0
     fputs(AFS_GOVERNMENT_MESSAGE, stdout);
